buildSrc/src/main/kotlin/VersionCodeGenerator.kt

```diff
@@ -19,20 +16,37 @@ import java.time.temporal.ChronoUnit
  *
  */
 
-fun String.getVersionCode(): Int {
-  // the date when the automatic version code generation started
-  val lastDate = LocalDate.of(2024, 7, 17)
+import java.time.LocalDate
+import java.time.temporal.ChronoUnit
 
-  // Calculate the number of days between the lastDate and today's date.
-  // This gives us the total number of days since the last version code was set.
-  val daysDifference = ChronoUnit.DAYS.between(lastDate, LocalDate.now()).toInt()
+/**
+ * The date when the automatic version code generation started.
+ */
+const val LAST_DATE = "2024-07-17"
+
+/**
+ * Base version code. This is the version code of the last release uploaded to the Play Store.
+ * We use this as the starting point for generating new version codes automatically.
+ */
+const val BASE_VERSION_CODE = 231101
 
-  // Base version code. This is the version code of the last release uploaded to the Play Store.
-  // We use this as the starting point for generating new version codes automatically.
-  val baseVersionCode = 231101
+fun String.getVersionCode(): Int {
+  // Get the current date. If the "KIWIX_ANDROID_RELEASE_DATE" environment
+  // variable is set(in YYYY-MM-DD format).
+  // It uses the specified date to generate the APK version code.
+  // Otherwise, it generates the version code based on the current date.
+  // See https://github.com/kiwix/kiwix-android/issues/4120 for more details.
+  val currentDate = if (!System.getenv("KIWIX_ANDROID_RELEASE_DATE").isNullOrEmpty()) {
+    LocalDate.parse(System.getenv("KIWIX_ANDROID_RELEASE_DATE"))
+  } else {
+    LocalDate.now()
```

Comment:
debug message should be here making clear the value of the date and where it comes from

Reply:
I have added the log.